src/tests/ie_test_utils/common_test_utils/postgres_link.hpp

```diff
@@ -0,0 +1,62 @@
+// Copyright (C) 2022 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include "test_common.hpp"
+
+namespace CommonTestUtils {
+
+/// \brief Class-container for PostgreSQLLink-specific data, declared and implemented
+///        out of header. Definition mustn't be a part of this header.
+class PostgreSQLCustomData;
+
+/// \brief Class incapsulates logic for communication with PostgreSQL from test-side
+///        Most logic should be implemented in corresponding postgres_link.cpp
+class PostgreSQLLink {
+    /// \brief parentObject should store pointer to a parent object to be able
+    ///        access to some specific parent's fields, if needed. Isn't mandatory.
+    void* parentObject;
+    /// \brief customData contains internal object state. It mustn't be accessible
+    ///        from outside.
+    PostgreSQLCustomData* customData;
```

Comment:
smart pointer?

Reply:
Similar thing. I use it to reduce huge compilations, but smart pointers requires explicit declaration.

Reply:
I don't believe in OV we often use smart pointers in headers with forward declaration and all works well...

Reply:
It isn't question about a believing in something. :)

I think you asked about my motivation: why I didn't use smart pointers in some places. My motivation: reduce extra-dependencies and overcomplicating a code if it isn't necessary. Especially, when I was seeing different behavior using different compilers. Smart pointers have some documented limitations when they are using with incomplete types.

I decided to don't use them if I don't need to "forget about memory management". But it is allowed me to significantly reduce compilation and code complexity while developing.

Could most of parts refactored to use smart pointers? I think, yes. Is it necessary? That's why I asked to review a code :) If you have some rules/obligations to use smart pointers everywhere - I'm ok. I may refactor some parts, if everything else is OK, because I tired to wait 1.5-3 hours each time I change a comment in a header. :)

Here's a doc: https://en.cppreference.com/w/cpp/memory/shared_ptr
> std::shared_ptr may be used with an [incomplete type](https://en.cppreference.com/w/cpp/language/incomplete_type) T. However, the constructor from a raw pointer (template<class Y> shared_ptr(Y*)) and the template<class Y> void reset(Y*) member function may only be called with a pointer to a complete type (note that [std::unique_ptr](https://en.cppreference.com/w/cpp/memory/unique_ptr) may be constructed from a raw pointer to an incomplete type).

Here's an experiment of some author: https://howardhinnant.github.io/incomplete.html

Also you may find a lot of results by query "shared_ptr incomplete type".